src/serde/visited_nodes.rs

```diff
@@ -0,0 +1,96 @@
+/// This is a simple bitfield used to indicates whether a node has been visited
+/// during a tree search or not. We terminate a search path if we've reached the
+/// node first via a different (shorter) path.
+#[derive(Clone, Default)]
+pub struct VisitedNodes {
+    bits: Vec<usize>,
```

Comment:
I'd be tempted to change this to `u64`. But maybe `usize` really is better since you're careful to use `usize::BITS`.

Reply:
I didn't spent so much time considering this, but `u128` is probably inappropriate since most hardware registers aren't that wide. My understanding is that `usize` is some kind of "word size" for the machine. But perhaps rust doesn't really have implementation defined types like that either.

Reply:
I believe `usize` is intended to be the same width as pointers (which is why it's also good as the size of array indices).